File: Hackerrank/Data Structures/Linked List/insertNodeAtTail.c
SinglyLinkedListNode* insertNodeAtTail(SinglyLinkedListNode* head, int data) {
    SinglyLinkedListNode* node = (SinglyLinkedListNode*)malloc(sizeof(SinglyLinkedListNode));
    node->data = data;
    node->next = NULL;
    if (head == NULL) {
        head = node;
    }
    else {
        SinglyLinkedListNode* current = head;
        while (current->next != NULL) {
            current = current->next;
        }
        current->next = node;
    }
    return head; 
}

File: HackerRank/Data Structures/Linked List/insertNodeAtTail.cs
static SinglyLinkedListNode insertNodeAtTail(SinglyLinkedListNode head, int data) 
{
  SinglyLinkedListNode node = new(data);
  
  if (head == null) { 
      head = node;
      head.next = null;
  } else {
      SinglyLinkedListNode current = head; 
      while (current.next != null) {
          current = current.next;
      }
      current.next = node; 
  }
  
  return head;
}
